core/store/src/flat/types.rs

```diff
@@ -32,11 +38,43 @@ impl From<FlatStorageError> for StorageError {
     }
 }
 
+#[derive(BorshSerialize, BorshDeserialize, Debug)]
+pub enum FlatStorageStatus {
+    /// 'protocol_feature_flat_state' is disabled
+    Disabled,
+    Empty,
+    Creation(FlatStorageCreationStatus),
+    Ready(FlatStorageReadyStatus),
```

Comment:
Do we plan to add separate `StateSync` status here?
Looks like it's not necessary for MVP though, as during state sync node is doomed.

Reply:
Yes, my plan is to add an explicit `StateSync` status here in a separate PR.

Reply:
Let's also add brief explanation comments for each status.
And actually `'protocol_feature_flat_state' is disabled` is not correct, because it can be set if runtime is implemented by `KeyValueRuntime`. So it's better to say that flat storage is just not supported.